### PR TITLE
Api updates/cleanup 

### PR DESCRIPTION
**Pull request by Eugene-hu**
-Introduce args for bittensor objects in neurons
-Added a proper doc string for neurons and various bittensor classes
-All mocked objects are now separate from bittensor classes and in a separate file
-Removed the depricated Akatsuki network

### CODE REVIEW

File: bittensor/__init__.py [hunk 1]
```diff
@@ -43,18 +43,14 @@ def turn_console_off():
 # Substrate chain block time (seconds).
 __blocktime__ = 12
 
-__networks__ = [ 'local', 'nobunaga', 'akatsuki', 'nakamoto']
+__networks__ = [ 'local', 'nobunaga', 'nakamoto']
```
Comment (shibshib) [inline bittensor/__init__.py:46]: goodbye akatsuki!

File: bittensor/_axon/__init__.py [hunk 2]
```diff
@@ -34,7 +34,15 @@
...
-    """ Create and init Axon, whcih services Forward and Backward requests from other neurons.
+    """ The factor class for bittensor.Axon object
+    The Axon acts a grpc server for the bittensor network and allows for communication between neurons.
+    By default, the grpc server follows the bittensor protocol and transports forward and backwards requests
+    between validators and servers. 
+    
+    Examples:: 
+            >>> axon = bittensor.axon(config=config)
+            >>> subtensor = bittensor.subtensor(network='nakamoto')
+            >>> axon.serve(subtensor=subtensor)
```
Comment (shibshib) [inline bittensor/_axon/__init__.py:45]: Great job, this is great docustringing

File: bittensor/_metagraph/metagraph_impl.py [hunk 3]
```diff
@@ -429,7 +429,7 @@ def sync ( self, block: int = None, cached: bool = True ) -> 'Metagraph':
         logger.success(self.subtensor)
         if block == None:
             block = self.subtensor.get_current_block()
-            if cached and self.subtensor.network in ("nakamoto", "local"):
+            if cached and self.subtensor.network in ("nakamoto", "local", "endpoint"):
```
Comment (shibshib) [inline bittensor/_metagraph/metagraph_impl.py]: Should we do this? if we call an endpoint, what if it's a Nobunaga endpoint?

File: bittensor/_axon/__init__.py [hunk 4]
```diff
@@ -34,7 +34,15 @@
 from . import axon_impl
 
 class axon:
-    """ Create and init Axon, whcih services Forward and Backward requests from other neurons.
+    """ The factor class for bittensor.Axon object
```
Comment (unconst) [inline bittensor/_axon/__init__.py:37]: Awesome

File: bittensor/_metagraph/metagraph_mock.py [hunk 5]
```diff
@@ -117,144 +97,6 @@ def clear( self ) -> 'Metagraph':
         self._endpoint_objs = None
         return self
 
-    def forward (
```
Comment (unconst) [inline bittensor/_metagraph/metagraph_mock.py:120]: Very clean Eugene, nice

File: bittensor/_neuron/text/advanced_server/__init__.py [hunk 6]
```diff
@@ -31,10 +31,39 @@
 from .run import serve
 
 class neuron:
+    r"""
+    Creates a bittensor neuron that specializes in the serving. The advanced server miner
```
Comment (unconst) [inline bittensor/_neuron/text/advanced_server/__init__.py]: In the serving ?

File: bittensor/_wallet/wallet_mock.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                _mock (required=True, default=False):
+                    If true creates a mock wallet with random keys.
+        """
+        super().__init__(**kwargs)
+        # For mocking.
+        self._is_mock = _mock
+        self._mocked_coldkey_keyfile = None
+        self._mocked_hotkey_keyfile = None
+
+        print("---- MOCKED WALLET INITIALIZED- ---")
```
Comment (unconst) [inline bittensor/_wallet/wallet_mock.py:26]: ala will be happy

**Review by unconst — APPROVED** [review]
Great PR

File: bittensor/_neuron/text/advanced_server/__init__.py (hunk 6 above)
Comment (shibshib) [reply bittensor/_neuron/text/advanced_server/__init__.py]: the serving what?

File: bittensor/_neuron/text/advanced_server/run.py [hunk 8]
```diff
@@ -173,14 +186,15 @@ def time_check():
         else: 
             return False
```
Comment (shibshib) [inline bittensor/_neuron/text/advanced_server/run.py]: Don't think we need an else here, can just return that if it fails.

File: bittensor/_neuron/text/advanced_server/run.py [hunk 9]
```diff
@@ -36,19 +36,32 @@
...
     # Load/Sync/Save our metagraph.
-    metagraph = bittensor.metagraph ( 
-        subtensor = subtensor
-    ).load().sync().save()
+    if metagraph == None:
+        metagraph = bittensor.metagraph ( 
+            subtensor = subtensor
+        ).load().sync().save()
+    else: 
+        metagraph.load().sync().save()
```
Comment (shibshib) [inline bittensor/_neuron/text/advanced_server/run.py:64]: We don't really need an else here, let's practice just defaulting if the `if` is not satisfied.

File: bittensor/_neuron/text/template_server/run.py [hunk 10]
```diff
@@ -31,20 +31,34 @@
...
     # Load/Sync/Save our metagraph.
-    metagraph = bittensor.metagraph ( 
-        subtensor = bittensor.subtensor( config = config )
-    ).load().sync().save()
+    if metagraph == None:
+        metagraph = bittensor.metagraph ( 
+            subtensor = subtensor
+        ).load().sync().save()
+    else: 
+        metagraph.load().sync().save()
```
Comment (shibshib) [inline bittensor/_neuron/text/template_server/run.py]: Same as above.

File: bittensor/_subtensor/__init__.py [hunk 11]
```diff
@@ -102,7 +105,7 @@ def __new__(
         config.subtensor._mock = _mock if _mock != None else config.subtensor._mock
         if config.subtensor._mock == True or network == 'mock' or config.subtensor.network == 'mock':
```
Comment (shibshib) [inline bittensor/_subtensor/__init__.py:106]: all these `if __ OR __ OR __` still create room for erroneously returning a mock.. is there some way we can make it more stringent?

File: bittensor/_subtensor/subtensor_mock.py [hunk 12]
```diff
@@ -0,0 +1,163 @@
+from substrateinterface import SubstrateInterface
```
Comment (shibshib) [inline bittensor/_subtensor/subtensor_mock.py:1]: Nice job moving it out to its own class!

File: bittensor/_wallet/__init__.py [hunk 13]
```diff
@@ -65,12 +65,20 @@ def __new__(
...
+                path = config.wallet.path,
+                _mock = True,
+            )
+
+        else:
+            return wallet_impl.Wallet(
+                name = config.wallet.name, 
+                hotkey = config.wallet.hotkey, 
+                path = config.wallet.path,
+            )
```
Comment (shibshib) [inline bittensor/_wallet/__init__.py]: else not needed here.

File: bittensor/_wallet/wallet_mock.py (hunk 7 above)
Comment (shibshib) [reply bittensor/_wallet/wallet_mock.py:26]: 🎊🎊🎊🎊

File: bittensor/_metagraph/metagraph_impl.py (hunk 3 above)
Comment (Eugene-hu) [reply bittensor/_metagraph/metagraph_impl.py]: Nice catch! this change shouldn't be there.

File: bittensor/_neuron/text/advanced_server/__init__.py (hunk 6 above)
Comment (Eugene-hu) [reply bittensor/_neuron/text/advanced_server/__init__.py]: Oops, english is hard

File: bittensor/_neuron/text/advanced_server/run.py (hunk 8 above)
Comment (Eugene-hu) [reply bittensor/_neuron/text/advanced_server/run.py]: Fixed!

Comment (Eugene-hu) [reply bittensor/_neuron/text/advanced_server/run.py]: Fixed

**Pull request merged**